components/storage/indexeddb/mod.rs

```diff
@@ -408,46 +465,65 @@ impl IndexedDBManager {
+        let requested_version = version;
 
         // Step 4: Let db be the database named name in origin, or null otherwise.
         let (db_version, version) = match self.databases.entry(idb_description.clone()) {
             Entry::Vacant(e) => {
                 // Step 5: If version is undefined, let version be 1 if db is null, or db’s version otherwise.
                 // Note: done below with the zero as first tuple item.
 
-                // Step 6: If db is null, let db be a new database
-                // with name name, version 0 (zero), and with no object stores.
-                // If this fails for any reason, return an appropriate error
-                // (e.g. a "QuotaExceededError" or "UnknownError" DOMException).
-                // TODO: return error.
-                let engine =
-                    SqliteEngine::new(idb_base_dir, &idb_description, self.thread_pool.clone())
-                        .expect("Failed to create sqlite engine");
+                // https://www.w3.org/TR/IndexedDB-2/#open-a-database
+                // Step 6: If db is null, let db be a new database with name name, version 0, and no object stores.
+                // If creation or opening fails, return an appropriate error to the open request.
+                let engine = match SqliteEngine::new(
+                    idb_base_dir,
+                    &idb_description,
+                    self.thread_pool.clone(),
+                ) {
+                    Ok(engine) => engine,
+                    Err(err) => {
+                        let _ = sender.send(Err(backend_error_from_sqlite_error(err)));
+                        return;
+                    },
+                };
                 let created_db_path = engine.created_db_path();
                 let db = IndexedDBEnvironment::new(engine);
-                let db_version = db.version().expect("DB should have a version.");
+                let db_version = match db.version() {
+                    Ok(version) => version,
+                    Err(err) => {
+                        let _ = sender.send(Err(backend_error_from_sqlite_error(err)));
+                        return;
+                    },
+                };
 
                 let version = if created_db_path {
-                    version.unwrap_or(1)
+                    requested_version.unwrap_or(1)
                 } else {
-                    version.unwrap_or(db_version)
+                    requested_version.unwrap_or(db_version)
                 };
 
                 e.insert(db);
                 (db_version, version)
             },
             Entry::Occupied(db) => {
-                let db_version = db.get().version().expect("Db should have a version.");
+                let db_version = match db.get().version() {
+                    Ok(version) => version,
+                    Err(err) => {
+                        let _ = sender.send(Err(backend_error_from_sqlite_error(err)));
```

Comment:
here also a log for error. see                 debug!("Script exit during indexeddb database open");